File: backend/src/main/java/com/justweighit/Configuration.java
package com.justweighit;

import com.fasterxml.jackson.annotation.JsonProperty;
import io.dropwizard.client.HttpClientConfiguration;
import io.dropwizard.client.JerseyClientConfiguration;
import io.dropwizard.db.DataSourceFactory;

import javax.validation.Valid;
import javax.validation.constraints.NotNull;

public class Configuration extends io.dropwizard.Configuration {
	
	@JsonProperty("apiKey")
	private String apiKey;
	
	@Valid
	@NotNull
	@JsonProperty("jerseyClient")
	private JerseyClientConfiguration jerseyClient = new JerseyClientConfiguration();
	
	@Valid
	@NotNull
	@JsonProperty("database")
	private DataSourceFactory database = new DataSourceFactory();
	
	public DataSourceFactory getDataSourceFactory() {
		return database;
	}
	
	public JerseyClientConfiguration getJerseyClientConfiguration() {
		return jerseyClient;
	}
	
	public String getApiKey() {
		return apiKey;
	}
}

File: backend/src/main/java/com/justweighit/Configuration.kt
package com.justweighit

import com.fasterxml.jackson.annotation.JsonProperty
import io.dropwizard.Configuration
import io.dropwizard.client.JerseyClientConfiguration
import io.dropwizard.db.DataSourceFactory
import javax.validation.Valid
import javax.validation.constraints.NotNull

class Configuration : Configuration() {
    @JsonProperty("apiKey")
    val apiKey: String? = null

    @JsonProperty("jerseyClient")
    val jerseyClientConfiguration: @Valid @NotNull JerseyClientConfiguration? = JerseyClientConfiguration()

    @JsonProperty("database")
    val dataSourceFactory: @Valid @NotNull DataSourceFactory? = DataSourceFactory()
}